internal/cli/federatedauthentication/federationsettings/describe.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package federationsettings
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/usage"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231115010/admin"
+)
+
+const describeTemplate = `ID	IDENTITY PROVIDER ID	IDENTITY PROVIDER STATUS
+{{.Id}}	{{.IdentityProviderId}}	{{.IdentityProviderStatus}}
+`
+
+type DescribeOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	id    string
+	store store.FederationSettingsDescriber
+}
+
+func (opts *DescribeOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *DescribeOpts) Run() error {
+	params := &atlasv2.GetFederationSettingsApiParams{
+		OrgId: opts.id,
+	}
+	r, err := opts.store.FederationSetting(params)
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+// atlas federatedAuthentication federationSettings describe <orgID>.
+func DescribeBuilder() *cobra.Command {
+	opts := new(DescribeOpts)
+	cmd := &cobra.Command{
+		Use:     "describe <orgID>",
+		Aliases: []string{"get"},
+		Args:    require.ExactArgs(1),
+		Short:   "Return the Federation Settings details for the specified organization.",
+		Long:    fmt.Sprintf(usage.RequiredRole, "Organization Owner"),
+		Annotations: map[string]string{
+			"orgIDDesc": "Unique 24-digit string that identifies the organization.",
+			"output":    describeTemplate,
+		},
+		Example: `  # Return the JSON-formatted Federation Settings details for the organization with the ID 5e2211c17a3e5a48f5497de3:
+  atlas federatedAuthentication federationSettings describe 5e2211c17a3e5a48f5497de3 --output json`,
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			return opts.PreRunE(
+				opts.initStore(cmd.Context()),
```

Comment:
missing `opts.ValidateOrgID`?

Reply:
we can but this command takes the org ID explicitly, so it doesn't use the opts orgID. I looked at [other org-related ](https://github.com/mongodb/mongodb-atlas-cli/blob/master/internal/cli/organizations/delete.go#L71)commands for reference,

Reply:
yes, I thought the same and decided this approach becasue of how atlas organizations and atlas projects work, if you do atlas projects describe it complains and ask for the ID: 

```shell
 atlas projects describe
Error: "atlas projects describe" requires 1 argument, received 0

Usage:  atlas projects describe <ID> [flags]
```

Reply:
the design principle that we generally use is [this](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md#adding-a-new-command)

> Commands follow a [RESTful](https://en.wikipedia.org/wiki/Representational_state_transfer) approach to match the APIs, whenever possible. For that reason, command arguments tend to match the path and query params of the APIs, with the last param being a required argument and the rest handled via flag options.

I think we could leave it as arg but make it optional.